flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -377,4 +424,9 @@ private void assertLatestWatermarkAlignmentEvent(int subtask, long expectedWater
         assertThat(events.get(events.size() - 1))
                 .isEqualTo(new WatermarkAlignmentEvent(expectedWatermark));
     }
+
+    private void assertHasNoMoreSplits(int subtask, boolean expected) {
+        sourceCoordinator.getContext().signalNoMoreSplits(0);
```

Comment:
```suggestion
        sourceCoordinator.getContext().signalNoMoreSplits(subtask);
```

We should use subtask instead of 0 here.